src/beacon.ts

```diff
@@ -129,23 +137,40 @@ class Beacon {
   }
 }
 
-const createBeaconInstance = () => {
-  return (url: string, body: string, config?: BeaconConfig) => {
+/**
+ * @public
+ */
+export function createBeacon(init: BeaconInit = {}): {
+  beacon: BeaconFunc;
+  database: RetryDB;
+} {
+  const { beaconConfig, retryDBConfig } = prepareConfig(init);
+  const database = new RetryDB(retryDBConfig);
+  const beacon: BeaconFunc = (url, body) => {
     if (!supportFetch) {
       return;
     }
-    new Beacon(url, body, config);
+    new Beacon(url, body, beaconConfig, database);
   };
-};
-
-/**
- * @example
- * ```
- * import beacon from 'beacon-transporter';
- * beacon(`/api`, 'hi', { retryCount: 3 })
- * ```
- * @public
- */
-const beacon = createBeaconInstance();
+  return { beacon, database };
+}
 
-export default beacon;
+function prepareConfig(init: BeaconInit): Required<BeaconInit> {
+  const beaconConfig = init.beaconConfig || {
+    retry: {
+      limit: 0,
+    },
+  };
+  const retryDBConfig = init.retryDBConfig || {
+    storeName: 'default',
```

Comment:
I would use a different default name since isn't this a global name for everything on the domain?

Reply:
Actually objectStore name should not be customized by consumer, database name is. I changed in the latest commit.